app/lib/test_applications.rb

```diff
@@ -80,45 +89,66 @@ def self.create_application(states:, courses_to_apply_to: nil)
 
       application_choices
     end
+  ensure
+    travel_back
   end
 
   def self.put_application_choice_in_state(choice, state)
-    # This is only supposed to happen after 7 days, but SendApplicationToProvider
-    # doesn't check the `edit_by` date of the ApplicationChoice
+    travel_to(choice.edit_by) if choice.edit_by > Time.zone.now
     SendApplicationToProvider.new(application_choice: choice).call
-    choice.update(edit_by: Time.zone.now)
     return if state == :awaiting_provider_decision
 
     if state == :offer
```

Comment:
What is your view on case statements? I think this `if elseif else` might be clearer as one, and would stop the repetition of 'state'.

Reply:
Yes it's as good a candidate for a case statement as you are likely to see. Thanks!